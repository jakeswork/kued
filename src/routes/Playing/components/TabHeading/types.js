import { node, shape, bool } from 'prop-types';

export const propTypes = {
  children: node,
  classes: shape({}).isRequired,
  sub: bool,
};

export const defaultProps = {
  children: null,
  sub: false,
};
