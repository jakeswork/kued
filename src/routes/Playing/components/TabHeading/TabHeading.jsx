import React from 'react';
import { Tab } from 'react-tabs';

import { propTypes, defaultProps } from './types';
import Text from '../../../../components/Text';

const TabHeading = ({
  children,
  classes,
  sub,
  ...rest
}) => (
  <Tab className={classes.tab} {...rest}>
    <Text bold h4={!sub}>{ children }</Text>
  </Tab>
);

TabHeading.propTypes = propTypes;

TabHeading.defaultProps = defaultProps;

TabHeading.tabsRole = 'Tab';

export default TabHeading;
