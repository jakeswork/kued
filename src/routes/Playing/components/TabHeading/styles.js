export default {
  tab: {
    cursor: 'pointer',
    userSelect: 'none',
    marginRight: 32,
  },
};
