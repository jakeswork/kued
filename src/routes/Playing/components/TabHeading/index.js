import withStyles from 'react-jss';

import TabHeading from './TabHeading';
import styles from './styles';

export default withStyles(styles)(TabHeading);
