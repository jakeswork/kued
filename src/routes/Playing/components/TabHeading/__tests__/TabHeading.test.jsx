import React from 'react';
import { shallow } from 'enzyme';

import TabHeading from '../TabHeading';

const defaultProps = {
  classes: {},
};

describe('The TabHeading instance', () => {
  describe('When rendered with default props', () => {
    it('should render without throwing errors', () => {
      const wrapper = shallow(<TabHeading {...defaultProps} />);

      expect(wrapper).toMatchSnapshot();
    });
  });
});
