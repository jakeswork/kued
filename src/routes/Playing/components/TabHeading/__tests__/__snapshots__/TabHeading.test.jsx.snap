// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The TabHeading instance When rendered with default props should render without throwing errors 1`] = `
<Tab
  className="react-tabs__tab"
  disabledClassName="react-tabs__tab--disabled"
  focus={false}
  id={null}
  panelId={null}
  selected={false}
  selectedClassName="react-tabs__tab--selected"
>
  <Jss(Text)
    analyticsLabel={null}
    bold={true}
    caption={false}
    h1={false}
    h2={false}
    h3={false}
    h4={true}
    h5={false}
    link={null}
    style={Object {}}
  />
</Tab>
`;
